languages/scxml/src/main/java/de/jplag/scxml/parser/PeekAdapter.java

```diff
@@ -0,0 +1,47 @@
+package de.jplag.scxml.parser;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import de.jplag.scxml.ScxmlTokenType;
+import de.jplag.scxml.parser.model.StatechartElement;
+
+/**
+ * A parser adapter that provides a way to retrieve a list of token types. When a token is added, only the ordinal of
+ * its type is stored. This can be used to "peek" at a list of token types that are extracted when visiting a
+ * statechart.
+ */
+public class PeekAdapter extends ScxmlParserAdapter {
```

Comment:
You extend the parser adapter. Thus this class is still functionally a full parser adapter. The name and comment, however, do not reflect that. Could you elaborate on the design here?